### DISCUSSION TITLE
DS1302 - multiple devices on SPI bus

### DISCUSSION THREAD

**Discussion by Stredoslovak** [Makuna/Rtc#141] [Q&A] [answered]
Hello, I'm using a DS1302 in one of my projects and I need to expand the functionality of said project. It's built upon an ESP32 and already using SPI display and DS1302 chip and several other non-SPI related stuff. I'm running out of pins and I would like to add an SD card functionality to the project.

**PROLEM**
As of now, the RTC and display both use completely different pins for communication with the ESP. Which is fine but I need to merge all SPI communications onto the same pins (MISO, MOSI, CLK).
**QUESTION**
What I want to know is if it is possible to achieve using this library.
_Can I use the same pins for communication with other devices or do I need to have pins reserved for the DS1302?_

**EDIT**
After a bit of research I found out that the DS1302 is not using SPI protocol, in fact it uses some "silly" variation of it. 
So that pretty much sums up my question. - Basically it's much better to use the SPI version of the chip which are the DS1305/DS1306

**Comment by Makuna** [comment]
The DS1302 is not SPI, it is ThreeWire.  
SPI has Clock, CS (CE), Data Out (MOSI/COPI), and a Data In  (MISO/CIPO).
Three Wire has Clock, CE (CS), and Data (out and in).

EDIT:  You beat me to the answer.  DS3234 is another option and has excellent time keeping abilities with temperature compensation.

**Reply by Stredoslovak** [reply, marked as answer]
Thank you very much for a quick answer for what now seems to me like a stupid question.
I've added an edit that I've found out shortly after. 
But thanks anyways.
